core/analytics/src/main/java/com/moneymong/moneymong/analytics/CompositionEventTracker.kt

```diff
@@ -0,0 +1,7 @@
+package com.moneymong.moneymong.analytics
+
+import androidx.compose.runtime.staticCompositionLocalOf
+
+val LocalAnalyticsTracker = staticCompositionLocalOf<AnalyticsTracker> {
+    NoOpAnalyticsTracker()
+}
```

Comment:
```suggestion
val LocalAnalyticsTracker = staticCompositionLocalOf<AnalyticsTracker> {
    object : AnalyticsTracker {
        override fun logEvent(event: AnalyticsEvent) = Unit
    }
}
```

`NoOpAnalyticsTracker`클래스가 `staticCompositionLocalOf`의 초기화만을 위해서 사용된다면
위와 같이 하고 삭제해도 괜찮다 생각합니다!

Reply:
테스팅을 위한 Fake 객체 생성용입니다 !

Reply:
아하 글쿤요!
처음 코드 읽을 때 `NoOpAnalyticsTracker` 클래스의 역할이 모호하다 느껴졌는데요!
이후에 `LocalAnalyticsTracker`의 제네릭 타입을 `nullable` 하지 않게 만드시려고 생성한 의미가 없는 클래스라고 이해했습니다.

그러면 클래스로 빼는 것보다 `LocalAnalyticsTracker` 변수 초기화 과정에서 익명 객체로 더미값을 넣어주는 게 이해하기 편하다고 생각했었어요!!

Reply:
그런 방향으로 생각을 할 수도 있겠네요 ! `nullable`도 있겠지만, stub의 용도로 만든 것이 더 크다고 이해하시면 좋을 것 같아요.
익명 객체보다 `NoOpAnalyticsTracker`라는 클래스로 더미값을 넣는게 가독성 측면에서 더 좋다고 생각이 드는 것 같습니다.

Reply:
귣귣 머지 고고하시져~